### PR TITLE
`azurerm_network_security_group` - support for augmented security rules

### PR DESCRIPTION
**Pull request by svanharmelen**
This support was already added to the `azurerm_network_security_rule` resource in PR #692, but it wasn’t added to the `azurerm_network_security_group` resource at that time.

It did need a bit of additional logic to make it work properly in the `azurerm_network_security_group` resource as well, but it all worked out nicely.

All related unit- and acceptance tests were successfully run before creating this PR.

### CODE REVIEW

File: azurerm/resource_arm_network_security_group.go [hunk 1]
```diff
@@ -216,83 +249,174 @@ func resourceArmNetworkSecurityGroupDelete(d *schema.ResourceData, meta interfac
...
+		destination_address_prefix := sgRule["destination_address_prefix"].(string)
+		priority := int32(sgRule["priority"].(int))
+		access := sgRule["access"].(string)
+		direction := sgRule["direction"].(string)
+		protocol := sgRule["protocol"].(string)
+
+		properties := network.SecurityRulePropertiesFormat{
+			SourcePortRange:          &source_port_range,
+			DestinationPortRange:     &destination_port_range,
+			SourceAddressPrefix:      &source_address_prefix,
```
Comment (pablo-ruth) [inline azurerm/resource_arm_network_security_group.go]: SourceAddressPrefixes should be expanded too, if not during an update of an existing nsg we will drop enhanced rules.

File: azurerm/resource_arm_network_security_group.go [hunk 2]
```diff
@@ -216,83 +249,174 @@ func resourceArmNetworkSecurityGroupDelete(d *schema.ResourceData, meta interfac
...
+			DestinationAddressPrefix: &destination_address_prefix,
```
Comment (pablo-ruth) [inline azurerm/resource_arm_network_security_group.go]: Same thing for DestinationAddressPrefixes.

File: azurerm/resource_arm_network_security_group.go (hunk 1 above)
Comment (svanharmelen) [reply azurerm/resource_arm_network_security_group.go]: Could you explain what you mean exactly? I'm afraid I don't understand what you mean by `SourceAddressPrefixes should be expanded too`...

Comment (svanharmelen) [reply azurerm/resource_arm_network_security_group.go]: Maybe you missed that I check and set those fields (if needed) as well? See below starting from line 288...

Comment (pablo-ruth) [reply azurerm/resource_arm_network_security_group.go]: I'm sorry, I made a mistake during my test, your PR fix that problem with #692... And it works as described 👍

Comment (svanharmelen) [reply azurerm/resource_arm_network_security_group.go]: Check, cool...

**Comment by tombuildsstuff** [comment]
Tests pass:

screen shot 2018-02-28 at 17 53 01

**Comment by svanharmelen** [comment]
Thanks! And no problem...

**Comment by tombuildsstuff** [comment]
:wave: hey @svanharmelen  

Just to let you know that this has just been released in v1.2.0 of the AzureRM Provider - full details of what's included are available here: https://github.com/terraform-providers/terraform-provider-azurerm/blob/v1.2.0/CHANGELOG.md#120-march-02-2018

Thanks!

**Comment by svanharmelen** [comment]
Cool... Thanks for the heads up!

**Pull request merged**